src/js/utils.js

```diff
@@ -83,6 +83,21 @@ export async function httpRequest(fetchMethod, path, options = requestOptions) {
   }
 }
 
+export function fetchPages(path, { fetch, baseURL }, onDataReceived, onError) {
+  const localFetch = (resource) => {
+    httpGet(fetch, new URL(resource, baseURL), onSuccess, onError)
+  }
+  const onSuccess = ({ data, headers }) => {
+    const links = parseLinkHeader(headers.get('Link'))
+    const nextLink = Object.hasOwn(links, 'next') ? links.next[0] : null
+    onDataReceived(data, nextLink === null)
+    if (nextLink !== null) {
+      setTimeout(localFetch, 25, nextLink)
```

Comment:
Why after a timeout? Since this is onSuccess, the previous request has already completed. Is there a reason to wait a little longer on top of that?

Reply:
I think that I did this to avoid tight looping (eg, let the browser and the server breath a little). Not completely sure if it is necessary or not. I can't recall if there is a stack depth issue or not since `localFetch` results in `onSuccess` being called. I think that there is some async buried in there somewhere that prevents that though 😕 

This should also sever the closure chain that would otherwise result... _I think_

Reply:
I don't think there is a stack depth issue, `setTimeout` queues up the callback out of the main thread, so the enclosing `onSuccess` function finishes, and the `localFetch` callback gets executed outside of it.